src/test/ramsesCL.test.ts

```diff
@@ -0,0 +1,147 @@
+        INonfungiblePositionManager.abi,
+        pool.signer
+      );
+    });
+
+    beforeAfterReset({ beforeAll, afterAll, provider });
+
+    describe("Liquidity", () => {
+      it("approves unlimited USDC and WETH on for Velodrome CL", async () => {
+        await pool.approveSpender(RAMSES_POSITION_MANGER, USDC, MAX_AMOUNT);
+        await pool.approveSpender(RAMSES_POSITION_MANGER, WETH, MAX_AMOUNT);
+        const UsdcAllowanceDelta = await allowanceDelta(
+          pool.address,
+          USDC,
+          RAMSES_POSITION_MANGER,
+          pool.signer
+        );
+        await expect(UsdcAllowanceDelta.gt(0));
+      });
+
+      it("adds USDC and WETH to a Velodrome CL (mint position)", async () => {
+        const usdcBalance = await pool.utils.getBalance(USDC, pool.address);
+        const wethBalance = await pool.utils.getBalance(WETH, pool.address);
+        await pool.addLiquidityUniswapV3(
+          Dapp.RAMSESCL,
+          WETH,
+          USDC,
+          usdcBalance.div(2),
+          wethBalance.div(2),
+          null,
+          null,
+          -204460,
+          -193470,
+          500
+        );
+
+        tokenId = (
+          await ramsesPositionManager.tokenOfOwnerByIndex(pool.address, 0)
+        ).toString();
+        expect(tokenId).not.toBe(null);
+      });
+
+      it("increases liquidity in a CL position", async () => {
+        const usdcBalance = await pool.utils.getBalance(USDC, pool.address);
+        const wethBalance = await pool.utils.getBalance(WETH, pool.address);
+        const positionBefore = await ramsesPositionManager.positions(tokenId);
+        await pool.increaseLiquidity(
+          Dapp.RAMSESCL,
+          tokenId,
+          usdcBalance.div(2),
+          wethBalance.div(2)
+        );
+        const positionAfter = await ramsesPositionManager.positions(tokenId);
+        expect(positionAfter.liquidity.gt(positionBefore.liquidity));
+      });
+
+      it("decreases liquidity from a CL position", async () => {
+        const positionBefore = await ramsesPositionManager.positions(tokenId);
+        await pool.decreaseLiquidity(Dapp.RAMSESCL, tokenId, 50);
+        const positionAfter = await ramsesPositionManager.positions(tokenId);
+        expect(positionAfter.liquidity.lt(positionBefore.liquidity));
+      });
+
+      it("collects fess of a CL position", async () => {
+        await provider.send("evm_increaseTime", [24 * 3600 * 3]); // 1 day
+        await provider.send("evm_mine", []);
+        await pool.claimFees(Dapp.RAMSESCL, tokenId);
+        expect((await balanceDelta(pool.address, USDC, pool.signer)).gt(0));
+      });
+
+      it("get rewards of a CL position", async () => {
+        await provider.send("evm_increaseTime", [24 * 3600 * 3]); // 1 day
+        await provider.send("evm_mine", []);
+        await pool.getRewards(Dapp.RAMSESCL, tokenId, [RAM]);
+        expect((await balanceDelta(pool.address, RAM, pool.signer)).gt(0));
+      });
```

Comment:
we probably can add a 100% decreaseLiquidity test at the end.

Reply:
added test [f9f078e](https://github.com/dhedge/dhedge-v2-sdk/pull/63/commits/f9f078ed20ccc58cf1aaebc0c7b5db5d08ace2be)